### ISSUE TITLE
UBSDKLoginManager shows error 

### ISSUE THREAD

**Issue by akki6230** [uber/rides-ios-sdk#26]
I tried to login with that code which is give below - with or without using redirect_URI but it always show error orderly 
1.The Redirect URI provided did not match what was expected.
2.  Invalid Redirect URI provided. or The server was unable to understand your request. most of time its crashes.
even I tried the example code It shows same error. I checked my whole project but there is no any other issue I found. Please help me to solve my issue.
Thanks

    UBSDKLoginManager *loginManager = [[UBSDKLoginManager alloc]init];
    NSArray<UBSDKRidesScope *> *requestedScopes = @[ UBSDKRidesScope.RideWidgets, UBSDKRidesScope.Profile, UBSDKRidesScope.Places ];
    [loginManager loginWithRequestedScopes:requestedScopes presentingViewController:self completion:^(UBSDKAccessToken * _Nullable accessToken, NSError * _Nullable error) {
        if (accessToken) {
            //[self _showMessage:UBSDKLOC(@"Saved access token!")];
        } else {
            //[self _showMessage:error.localizedDescription];
        }
    }];

**Comment by faarwa** [comment]
Hi @akki6230, can you please verify if the Redirect URI configured in your app's `Info.plist` is the same as the one in your app settings on the [Uber Developers Dashboard](https://developer.uber.com/dashboard/)?

**Comment by akki6230** [comment]
Very Thanks @faarwa you replied :)  but I double checked both are same. :(

**Comment by faarwa** [comment]
You mentioned that most of the time it crashes - would you be able to provide the call stack for the crash?

**Comment by akki6230** [comment]
My redirect_URI format was wrong now it is not showing any error or crashes. Now it shows **The Ride Request Widget encountered a problem** ??

**Comment by jbrophy17** [comment]
Sorry to hear you're still having trouble @akki6230. That error message should only be presented by the LoginManager when there is no internet connection. The text is incorrectly based on being used from within the Ride Request Widget and will be updated soon. 

Can you confirm that the device you're testing on is connected to the internet?

**Comment by akki6230** [comment]
Yes it is connected to Internet :)

**Comment by jbrophy17** [comment]
Tough to tell without more information, but that message indicates a problem involving your access token (getting it, saving it, or the token itself). I would suggest clearing your access token (call `TokenManager.deleteToken()` before what you added from Quick integration) and trying again.

**Comment by akki6230** [comment]
Firstly 👍  for you both. :) for your instant help.
I tried but no success. Now it seems there is problem with my SDK Integration. So please at first let me clear about the redirect_URI - 
1. is redirect_URI is optional ? If it is then why it shows **Invalid Redirect URI provided. or The server was unable to understand your request.** 
2. Which format an UBER redirect_URI should have?? please mentation an example too. I'm sure I'm not adding a right redirect_URI.
Please help me as you done so far. 
Thanks

**Comment by jbrophy17** [comment]
1. The redirect uri is not optional, you need to provide it for implicit grant login. 
2. The format can be any URL really, but I would suggest using a unique scheme for your app, so for example "myApp://connect". And "myApp://connect" is what you have in the developer dashboard AND in your plist (they must match)

**Comment by akki6230** [comment]
Very Sorry I forgot to tell you that my problem was fixed. When I set **UberRides.xcodeproj target/General/Linked Frameworks and Libraries - ObjectMapper.framework status to Required earlier it was Optional**, set by me cause  I was getting error something like **Library not loaded ... image not found**. but as I set it to Required again and added ObjectMapper.framework to My Projects/General/Embeded Binaries. Everything worked.
![screen shot 2016-04-28 at 11 50 57 am](https://cloud.githubusercontent.com/assets/12725673/14947481/d90fc27e-1051-11e6-8203-c6f843c4ebdc.png)
 
Thanks for your support too
